File: app/src/main/java/com/example/anchornotes/data/db/NoteWithTags.java
package com.example.anchornotes.data.db;

import androidx.room.Embedded;
import androidx.room.Junction;
import androidx.room.Relation;
import java.util.List;

public class NoteWithTags {
    @Embedded public NoteEntity note;

    @Relation(
            parentColumn = "id",
            entityColumn = "id",
            associateBy = @Junction(
                    value = NoteTagCrossRef.class,
                    parentColumn = "noteId",
                    entityColumn = "tagId"
            )
    )
    public List<TagEntity> tags;
}

File: app/src/main/java/com/example/anchornotes/data/db/NoteWithTags.kt
package com.example.anchornotes.data.db

import androidx.room.Embedded
import androidx.room.Relation

/**
 * NoteEntity with its associated tags.
 * Used for search results that need tag information.
 */
data class NoteWithTags(
    @Embedded
    val note: NoteEntity,
    
    @Relation(
        parentColumn = "id",
        entityColumn = "id",
        associateBy = androidx.room.Junction(
            value = NoteTagCrossRef::class,
            parentColumn = "noteId",
            entityColumn = "tagId"
        )
    )
    val tags: List<TagEntity> = emptyList()
)

